docs/sql-ref-datatypes.md

```diff
@@ -49,6 +49,44 @@ Spark SQL and DataFrames support the following data types:
   absolute point in time.
   - `DateType`: Represents values comprising values of fields year, month and day, without a
   time-zone.
+* Interval types
+  - `YearMonthIntervalType(startField, endField)`: Represents a year-month interval which is made up of a contiguous subset of the following fields:
+    - MONTH, months within years `[0..11]`,
+    - YEAR, years in the range `[0..178956970]`.
+
+    Individual interval fields are non-negative, but an interval itself can have a sign, and be negative.
+
+    `startField` is the leftmost field, and `endField` is the rightmost field of the type. Valid values of `startField` and `endField` are 0(MONTH) and 1(YEAR). Supported year-month interval types are:
+    
+    |Year-Month Interval Type|Short form|An instance of the type|
```

Comment:
do we really need to mention the short form? It seems only available in Scala `YearMonthIntervalType.apply`

Reply:
You are slightly late. I added the `SQL type` column instead. I think it is more important.